### ISSUE TITLE
Design System Staging Review - Accessibility Feedback - Maintenance Banner web component

### ISSUE THREAD

**Issue by bethdeconinck** [department-of-veterans-affairs/va.gov-team#61024]
## VFS actions
- [ ] Assign this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [ ] Comment on this ticket:
  - [ ] If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - [ ] When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - [ ] When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [ ] Close the ticket when all feedback has been addressed.

## Thoughts/questions

- One thing that's tricky about this one is uncertainty around where the banner will appear in the DOM. Looking at VA.gov, I did find an empty `div` with an `aria-label` of "Maintenance banner" that's between the `header` and the `main` elements. I'm going to assume that's where this gets dropped in? If not, this feedback should all still be true but might need to be adjusted slightly.

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

- **Must:** Remove `role="banner"`. In general, each page should only have one element with a role of `banner`, and when a `header` is a direct child of the `body` element it gets that role by default. The spec for `banner` doesn't quite fit this use case anyway, so keeping it here might be confusing to screen reader users navigating by landmark. 
   - I don't *think* there's another role that needs to be added in here either, but there might be a case to be made that the error-style banner should have `role="alert"` to notify users that maintenance is in progress and may impact what they're doing. But I feel like this is more of an advisory than a critical notification and probably isn't worth the disruption for screen reader users.
- **Must:** Right now, the heading on the banner is coded as an `h4`. For the placement between the `header` and the `main`, an `h2` would be a better fit.
   - I think hard-coding as an `h2` is fine, but if there's any chance this banner might show up somewhere else consider adding a property for setting heading level for greater flexibility.

**Comment by jeana-adhoc** [comment]
`role="banner"` was removed [in this commit](https://github.com/department-of-veterans-affairs/component-library/commit/9c098425c332af96d69faa5f130758e3a0c157e3#diff-4a54222549d0849e45157773b39ea4b27758c7bbe31fe8d4f11d144fdfc4cfee) in 2024.

However, the `<h4>` is still hardcoded, and should be an `<h2>` by default, but it would be even better to allow teams to define a different heading level.
